src/k8s/pkg/k8sd/features/cilium/network_test.go

```diff
@@ -0,0 +1,179 @@
+package cilium_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/canonical/k8s/pkg/client/helm"
+	helmmock "github.com/canonical/k8s/pkg/client/helm/mock"
+	"github.com/canonical/k8s/pkg/k8sd/features/cilium"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/snap"
+	snapmock "github.com/canonical/k8s/pkg/snap/mock"
+	"github.com/canonical/k8s/pkg/utils"
+	"k8s.io/utils/ptr"
+)
+
+// NOTE(hue): status.Message is not checked sometimes to avoid unnecessary complexity
+
+func TestDisabled(t *testing.T) {
+	t.Run("HelmApplyFails", func(t *testing.T) {
+		applyErr := errors.New("failed to apply")
+		helmM := &helmmock.Mock{
+			ApplyErr: applyErr,
+		}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+			},
+		}
+		cfg := types.Network{
+			Enabled: ptr.To(false),
+		}
+
+		status, err := cilium.ApplyNetwork(context.Background(), snapM, cfg, nil)
+
+		g := NewWithT(t)
```

Comment:
We put this after t.Run

Reply:
right, but is it any different? I definitely can change that, but I feel like this shouldn't really matter. I mean if I have to choose I'd personally prefer keeping "gomega" things and "expects" in the same place. or does it make a real difference? I'm super welcoming to nit and repo/team specific code style comments but I feel like this might be too fine grained for that matter. wdyt?

Reply:
Well it matters because these tests would be the only ones that don't stick to the (unwritten) convention we have established. I definitely see what you're going for with keeping variable declarations close to their use, but I would prefer if we stick to the style we have.